aten/src/ATen/native/TensorShape.cpp

```diff
@@ -28,13 +28,31 @@ static void check_cat_no_zero_dim(TensorList tensors) {
 Tensor & cat_out(Tensor & result, TensorList tensors, int64_t dim) {
   check_cat_no_zero_dim(tensors);
   dim = legacy_cat_wrap_dim(dim, tensors);
-  return at::_cat_out(result, tensors, dim);
+  return at::_cat_out(result, tensors, dim, false, 0);
 }
 
 Tensor cat(TensorList tensors, int64_t dim) {
   check_cat_no_zero_dim(tensors);
   dim = legacy_cat_wrap_dim(dim, tensors);
-  return at::_cat(tensors, dim);
+  return at::_cat(tensors, dim, false, 0);
+}
+
+Tensor & cat_out(Tensor & result, TensorList tensors, int64_t dim, Scalar pad_value) {
+  check_cat_no_zero_dim(tensors);
+  dim = legacy_cat_wrap_dim(dim, tensors);
```

Comment:
we shouldn't do this legacy_cat_wrap_dim stuff if `pad_value` is specified; it is behavior we needed before we had n-dimensional empty tensors, but we have those now.  Let's not extend legacy semantics to new functionality.